src/algebra/pointwise.lean

```diff
@@ -578,6 +578,11 @@ section monoid
   multiplication `*` as "multiplication". -/
 @[derive inhabited] def set_semiring (α : Type*) : Type* := set α
 
+/-- The natural order by inclusion on subsets of `α` is available on `set_semiring α`. -/
+attribute [derive partial_order] set_semiring
+/-- The natural order by inclusion on subsets of `α` has `∅` as its bottom element. -/
+attribute [derive order_bot] set_semiring
+
 /-- The identitiy function `set α → set_semiring α`. -/
 protected def up (s : set α) : set_semiring α := s
```

Comment:
It would be nice to add the two lemmas
```lean
@[simp] lemma up_le_up {s t : set α} : up s ≤ up t ↔ s ⊆ t := iff.rfl
@[simp] lemma down_subset_down {s t : set_semiring α} : down s ⊆ down t ↔ s ≤ t := iff.rfl

Reply:
That is a great suggestion!  I will do it in a separate PR: this PR is blocking another one and I would like this one to get merged.  Does this sound ok?

Reply:
I'd argue it belongs in this one: these lemmas are the ones that tell the reader what the new instance means

Reply:
Ok, I'll add them, but it may have to wait until after the weekend: not sure I'll have time today!

Reply:
I pushed the change myself from mobile :)

Reply:
That works too, thanks!  I don't feel confident enough to type lean from mobile without checking!

Reply:
Seems I got it wrong too!